src/lib/ModalState.ts

```diff
@@ -249,7 +249,7 @@ const createModalState = (): ModalStateType<any> => {
     action,
     callback,
     modalName,
-  }: ModalStatePendingClosingAction): ModalPendingClosingAction | null => {
+  }: ModalStatePendingClosingAction): ModalPendingClosingAction | (() => void) | void | null => {
```

Comment:
just for me to understand: why would the new returned type be `ModalPendingClosingAction | (() => void) | void | null` instead of just `ModalPendingClosingAction  | null | void` if we're invoking the function during the `return`?

Reply:
Because callback?.() return (() => void)

Reply:
@antonRoyenko i think i may be missing something but: if we write `return callback?.()`, wouldn't it directly return `void` instead of `() => void`  since modalfy would have already invoked the function just then?

`return callback?.()` would only return `() => void` if someone provided modalfy with function _that returns_ a function itself as a callback, which would not make sense since they'd have to invoke that returned function callback themselves somehow.

i'd be curious to get your point of view regarding my reasoning.

Reply:
oh yes, omitted that, I agree with you

Reply:
hey @antonRoyenko! did you get the opportunity to get back to this pr to push the small changes we talked about, by any chance?